### ISSUE TITLE
One missing device breaks the whole integration

### ISSUE THREAD

**Issue by mikelward** [BottlecapDave/HomeAssistant-OctopusEnergy#1086]
### Describe the bug

I had to delete my electric car from my Octopus account to fix a password issue. 

This seems to have caused the whole integration to fail to start up. I'm missing unrelated sensors e.g. off peak and current tariff.

### Reproduction steps

Delete a smart device from Octopus account

### Expected behaviour

The integration marks the device as missing and otherwise works.

### Tariff Code

0

### Integration Version

13.1.2

### Home Assistant Version

2024.11.2

### Fresh Install?

After upgrading

### Home Assistant Logs

```
Logger: homeassistant.config_entries
Source: config_entries.py:635
First occurred: 19 November 2024 at 16:32:03 (2 occurrences)
Last logged: 09:07:49

Error setting up entry CENSORED for octopus_energy
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 635, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/__init__.py", line 143, in async_setup_entry
    await async_setup_dependencies(hass, config)
  File "/config/custom_components/octopus_energy/__init__.py", line 399, in async_setup_dependencies
    await async_save_cached_intelligent_device(hass, account_id, intelligent_device)
  File "/config/custom_components/octopus_energy/storage/intelligent_device.py", line 29, in async_save_cached_intelligent_device
    await store.async_save(intelligent_device.to_dict(omit_id=False))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'to_dict'
```

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by mikelward** [comment]
Fixed by 7e7290a?

**Comment by BottlecapDave** [comment]
Hello and sorry you're having issues. The reported error should be fixed as part of v13.1.3. Please upgrade to that and let me know if you're still having issues.
